### ISSUE TITLE
Add Speech to text

### ISSUE THREAD

**Issue by ms10398** [fossasia/susi_firefoxbot#15]
**Actual Behaviour**

Till now Speech to text is not implemented in the extended

**Expected Behaviour**

After this speech to text option will be there in the extension

**Would you like to work on the issue?**

Yes I would like to work on the issue

**Comment by tstreamDOTh** [comment]
How are you planning to implement this?

**Comment by ms10398** [comment]
I am thinking to use Webspeech API

**Comment by tstreamDOTh** [comment]
@ms10398 Web speech API is in an experimental stage. It is not advised to use it. Find out if there is a better way or else we can put this on hold for a while.

**Comment by ms10398** [comment]
I think its been a few year even google uses web speech API for thier google search
According to me its the best way to go for STT(Speech to Text) @tstreamDOTh

**Comment by ms10398** [comment]
Can you suggest any other better way @tstreamDOTh

**Comment by tstreamDOTh** [comment]
@ms10398 We are talking about web speech API for firefox and not chrome. Even I didn't find any appropriate resource to deal this. Anyway if you can implement it bugfree go ahead.

**Comment by ms10398** [comment]
Okay I will try to implement it @tstreamDOTh

**Comment by ms10398** [comment]
@harshit98 I am trying to implement it will try to produce the results soon

**Comment by harshit98** [comment]
Ok. @ms10398 If you face any difficulties, please let the co-developers know. :)

**Comment by ms10398** [comment]
Yes sure @harshit98

**Comment by tstreamDOTh** [comment]
@ms10398 Any updates?

**Comment by ms10398** [comment]
Sorry I couldn't put in that much time coz of exams will post the update soon as now exams are over.

**Comment by tstreamDOTh** [comment]
@ms10398 If this is not feasible we can drop this. Text to speech can be implemented but speech to text through extension doesn't seem feasible.

**Comment by tstreamDOTh** [comment]
@ms10398 I checked how it is implemented in susi_chromebot. There are bugs in it and there is a scope for improvement in it. Let us see how this can be done. Also look for alternatives for firefox.

**Comment by tstreamDOTh** [comment]
@harshit98 Any feedback or suggestions?

**Comment by ms10398** [comment]
Yes I agree @tstreamDOTh I am seeing all options and will try to implement the best I can make out.

**Comment by harshit98** [comment]
@tstreamDOTh I will check if something can be done on this.

**Comment by harshit98** [comment]
@tstreamDOTh @ms10398 I tried to replicate the feature as it is implemented in susi_chromebot. Currently, I'm observing microphone is not working even in susi_chromebot. I'm getting this error: 

![screen shot 2017-10-20 at 11 23 29 am](https://user-images.githubusercontent.com/22245418/31807060-754ae898-b589-11e7-8ae9-133b42c16cc4.png)

You guys can refer Susper files where STT is working fine.
-  <a href="https://github.com/fossasia/susper.com/blob/master/src/app/speechtotext/speechtotext.component.ts">speech-to-text component</a>
- <a href="https://github.com/fossasia/susper.com/blob/master/src/app/services/speech-synthesis.service.ts">speech-to-text file which enables the feature</a>

**Comment by ms10398** [comment]
You should check again @harshit98 It works in susi chromebot

**Comment by gabru-md** [comment]
@harshit98 please check once again. The microphone in susi_chromebot seems to be working.

**Comment by tstreamDOTh** [comment]
This is a GCI task. Anyone participating in GCI can take this up.

**Comment by tstreamDOTh** [comment]
@mariobehling Please add it in the GCI-tasks
